reconcile/utils/terrascript_aws_client.py

```diff
@@ -4514,32 +4513,37 @@ def _get_commit_sha(self, repo_info: Mapping) -> str:
         return ""
 
     def _get_asg_image_id(
-        self, image: Mapping, account: str, region: str
-    ) -> Tuple[Optional[str], str]:
+        self, filters: Iterable[Mapping[str, Any]], account: str, region: str
+    ) -> Optional[str]:
         """
         AMI ID comes form AWS Api filter result.
         AMI needs to be shared by integration aws-ami-share.
-        AMI needs to be taged with a tag_name and
-        its value need to be the commit sha comes from upstream repo.
+        AMI needs to be taged either with:
+          * a tag_name and its value need to be the commit sha comes from upstream repo.
+          * tag_name and value
         """
-        commit_sha = self._get_commit_sha(image)
-        tag_name = image["tag_name"]
+        tags: list[AmiTag] = []
+        for f in filters:
+            if f["provider"] == "git":
```

Comment:
It may be a bit too protective, but in our current model, does it make sense that we have two `git` providers? Would that break things? Should we limit it somehow?

Reply:
Allowing multiple git providers won't break anything but doesn't make absolute sense.

Reply:
I don't see a need to limit it

Reply:
fair enough.